usecases/vessel-detection/index.md

```diff
@@ -0,0 +1,211 @@
+conn = conn.authenticate_oidc()
+```
+
+### 2. Pre-processing prep
+
+This process graph requires either a sea or land mask to remove the land from our AOI. A seamask has been prepared for this example and is available on github. It was derived using the [Water Bodies](https://land.copernicus.eu/content/corine-land-cover-nomenclature-guidelines/html/) classes from Corine Land Cover.
+
+
+```python
+import folium
+
+# We are going to use the following GeoJson to operate as a sea mask for out process graph.
+SEA_MASK = "https://raw.githubusercontent.com/SerRichard/sea_mask/main/sea-mask-4326.json"
+
+fig = folium.Figure(width=600, height=400)
+map = folium.Map(location=[44.465488, 12.602316], zoom_start=8)
+folium.GeoJson(SEA_MASK).add_to(map)
+fig.add_child(map)
+```
+![SeaMask to be used](./seamask.png)
+
+### 3. Defining a process graph to detect vessels in our AOI.
+
+The spatial_extent we have defined includes, but is not limited to the sea mask that we will be using. We will use both available polarizations from the SENTINEL1_GRD collection, and run this graph on a little over a weeks worth of data.
+
+
+```python
+spatial_extent  = {
+          "west": 12.194377989297493,
+          "east": 12.758093271633888,
+          "south": 44.24420099164355,
+          "north": 44.85455845353388
+        }
+
+temporal_extent = [
+          "2021-10-01",
+          "2021-10-09"
+        ]
+
+collection      = "SENTINEL1_GRD"
+bands           = ["VV","VH"]
+
+s1_datacube = conn.load_collection(
+    collection,
+    spatial_extent=spatial_extent,
+    bands=bands,
+    temporal_extent=temporal_extent
+)
+```
+
+Load the geometries as a vector cube.
+
+
+```python
+sea_mask = s1_datacube.process("load_vector_cube", {"URL": SEA_MASK})
+```
+
+Replace the values that lie outside of our polygon with NaN values.
+
+
+```python
+masked_data = s1_datacube.mask_polygon(sea_mask)
+```
+
+Apply a window to our datacube, and per pixel, update the pixel's value to be the mean of the window surrounding it.
+
+This would be the equivalent of the `data.rolling()` from the xarray example above.
+
+
+```python
+aggregated_window = masked_data.aggregate_spatial_window(size=[30, 30], reducer="mean")
+```
+
+Apply the multiplication of our given value to the result of the spatial window. We multiply to increase the contrast between sea and ship, which makes the detection more reliable.
+
+
+```python
+multiplied = aggregated_window.apply(lambda x: x.multiply(4) )
+```
+
+We then compare the pixel values from the previous node, against the values of the data we initially masked. We want to keep any values from the initially masked data, in order to generate a boolean mask, which should reflect our vessels.
+
+I could not find a nice way to express this in the client, without it re-including the first three nodes of the process graph. So I have applied it directly as a process here.
+
+
+```python
+lt_comparison = multiplied.process("apply", {
```

Comment:
I think this should be replaced with merge_cubes(multiplied,masked_data,overlap_resolver="lt")
or else merge_cubes creating 2 separate bands, and then using an 'if' process in an apply_dimension on the bands.

Reply:
you probably got that repetition because of trying to put a datacube in the context. This construction however does not (or should not) give the result you are after. What will happen inside the apply callback is that each value in the cube is compared to a full datacube, which will/should give an error.
That is why you want to do merge_cubes _instead_ of an apply operation, so the 'lt' callback either has to go in merge_cubes as overlap resolver, or has to come after it as an extra apply_dimension.

Reply:
Context allows for a DataType of Any? It just seems a bit odd to me, but will give it a go!

Reply:
No worries, need to patch a bug in the apply_kernel for processes dask, but can produce the same output, will switch to using that process.

Reply:
This has been updated to use merge cubes and the overlap resolver